frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts

```diff
@@ -853,6 +874,31 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
             },
         ],
 
+        dataBufferedUntilPlayerTime: [
+            (s) => [s.dataBufferedUntilTimestamp, s.sessionPlayerData],
+            (dataBufferedUntilTimestamp, sessionPlayerData) => {
+                return Math.max(0, (dataBufferedUntilTimestamp ?? 0) - (sessionPlayerData?.start?.valueOf() ?? 0))
+            },
+        ],
+
+        /**
+         * we want a signal we can use to check if we should try to load more data when progressive loading
+         * if this is negative then it means the player time is past the current buffer -> we should load
+         * if it is within some tolerance -> we should load
+         */
+        percentPlayedOfBuffered: [
```

Comment:
> im not too familiar with buffer strategy but wondering out loud if we're concerned at all about a threshold being a % (4 hour recording, hitting 67% played of 2 hours buffered still leaves me with ~40 watchable minutes, do we need to load more)

@ksvat i just needed a hook to say when to load the next block of data (there might be a better hook!) and then had to pick _some_ threshold

since not loading was such a pain, i wanted to be slightly over eager so that we're more likely to load data we won't watch than not load data when we should

i was doing it based on time (e.g. 2 minutes or less till buffer) but it was confusing me when it didn't work

that might only be the cache busting problem, since I didn't figure that out till i'd got to a place i was sure this should be working